File: app/src/main/java/com/iaruchkin/deepbreath/network/endpoints/AqiEndpoint.java
package com.iaruchkin.deepbreath.network.endpoints;

import com.iaruchkin.deepbreath.network.dtos.AqiResponse;

import androidx.annotation.NonNull;
import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface AqiEndpoint {

    @GET("/feed/{location}/")
    Single<AqiResponse> get(@Path("location") @NonNull String section);
}

File: app/src/main/java/com/iaruchkin/deepbreath/data/aqi/remote/AqiEndpoint.kt
package com.iaruchkin.deepbreath.data.aqi.remote

import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Aqi api of api.waqi.info
 */
interface AqiEndpoint {
    @GET("/feed/{location}/")
    suspend fun get(
        @Path("location") section: String
    ): AqiResponse
}